src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -164,22 +164,39 @@ public static double BitIncrement(double x)
 
         public static unsafe double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask 
+                = unchecked((long)0b_1000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000);
 
-            var xbits = BitConverter.DoubleToInt64Bits(x);
-            var ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.IsSupported)
+            {
+                // Create vectors of the elements, and make them float to allow using SSE rather than SSE instructions
+                var xvec = Vector128.CreateScalarUnsafe(x).AsSingle();
+                var yvec = Vector128.CreateScalarUnsafe(y).AsSingle();
 
-            // If the sign bits of x and y are not the same,
-            // flip the sign bit of x and return the new value;
-            // otherwise, just return x
+                // Remove the sign from x, and remove everything but the sign from y
+                // Creating from a 'const long' is better than from the correct 'const float/double'
+                xvec = Sse.And(xvec, Vector128.CreateScalarUnsafe(~signMask).AsSingle());
```

Comment:
Use `ANDNPS` instead of creating 2 distinct masks?

Reply:
:+1: done